### DISCUSSION TITLE
Сonnection between Strongswan and CheckPoint

### DISCUSSION THREAD

**Discussion by smgroupit** [strongswan/strongswan#843] [Q&A] [answered]
There is a tunnel between Strongswan and CheckPoint. Strongswan is behind NAT. Router is configured to forward ports 500/UDP and 4500/UDP to the Ubuntu 20.04.3 LTS virtual machine with Strongswan 5.8.2.

The CheckPoint settings are as follows:

- Ecnryption Method - IKE v2 only
- IKE Security Phase 1: 
- Key exchange encryption - AES-256
- Data integrity with SHA-256
- IKE Security Phase 2: 
- IPsec data encryption - AES-256
- Data integrity - SHA-256
- VPN Tunnel Sharing - One VPN tunnel per each pair of hosts (ESP)
- Use shared secret.
- IKE (Phase 1) use Diffie-Hellman group - Group 14 (2048 bit) 
- Renegotiate IKE security - 1440 minutes
- IPsec (Phase 2)
- Use perfect Forward secrecy, use Difie-Hellman group - Group 14 (2048 bit) 
- Renegotiate IKE security - 1440 seconds
- Disable NAT inside the VPN community

Strongswan Settings:

conn office
  keyexchange=ikev2
  type=tunnel
  leftauth=psk
  left=172.16.52.84           #Host internal ip address
  #left=%defaultroute
  leftid=X.X.X.X
  leftsubnet=172.16.152.0/24  #Host externa ip/Encryption domain
  right=Y.Y.Y.Y         #Peer2 ip address
  rightauth=psk
  rightsubnet=10.22.12.0/23 #Peer2 accesible intranet
  #Phase 1 integrity check algos
  ike=aes256-sha256-modp2048!
  #Phase 2 Encryption algos
  esp=aes256-sha256-modp2048!
  ikelifetime=86400s
  lifetime=1440s
  reauth=yes
  rekey=yes
  keyingtries=%forever
  dpddelay=30
  dpdtimeout=120
  dpdaction=restart
  auto=start

An error occurs when trying to connect:

Jan 11 12:58:37 router charon: 07[IKE] initiating IKE_SA office[1] to Y.Y.Y.Y
Jan 11 12:58:37 router charon: 07[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
Jan 11 12:58:37 router charon: 07[NET] sending packet: from 172.16.52.84 [500] to Y.Y.Y.Y[500] (2164 bytes)
Jan 11 12:58:38 router charon: 13[NET] received packet: from Y.Y.Y.Y [500] to 172.16.52.84 [500] (832 bytes)
Jan 11 12:58:38 router charon: 13[ENC] payload of type NOTIFY more than 20 times (21) occurred in current message
Jan 11 12:58:38 router charon: 13[IKE] message verification failed

What could be the problem?

**Comment by tobiasbrunner** [comment]
Apparently, the responder sent back more notify payloads than the current limit of 20 per message allows. You can try increasing the limit here: https://github.com/strongswan/strongswan/blob/85d626e9aef9f12233fe79ea0b6fb61c89ff149d/src/libcharon/encoding/message.c#L44

**Reply by smgroupit** [reply]
Thanks. Is there a setting in Strongswan itself to increase the limit? Or do I need to rebuild it from the source code?

Now I have a Strongswan from the Ubuntu 20.04 LTS operating system repository.

**Reply by tobiasbrunner** [reply]
No, you have to rebuild from sources.

**Reply by smgroupit** [reply]
Thank you, I will try.

**Reply by tobiasbrunner** [reply]
Once you know what types of notifies they are (you could try to increase the log level for _enc_ to see that even now), there might be configuration options to avoid some of them.

**Comment by smgroupit** [comment, marked as answer]
The problem was solved by switching to the IKEv1 protocol.
